llms/mlx_lm/utils.py

```diff
@@ -157,19 +158,19 @@ def generate(
             tic = time.time()
         tokens.append(token.item())
 
-        if verbose:
-            s = tokenizer.decode(tokens)
+        s = tokenizer.decode(tokens)
+        if REPLACEMENT_CHAR not in s:
             if formatter:
                 formatter(s[skip:], prob.item())
-                skip = len(s)
-            elif REPLACEMENT_CHAR not in s:
+            else:
                 print(s[skip:], end="", flush=True)
-                skip = len(s)
+            skip = len(s)
```

Comment:
We don't want to print here in general. The API shouldn't print anything when `verbose=False`